File: CsLibrary/CsLibrary/Integer.cs
using System;

namespace JavaClasses {

    /// <summary>
    /// The Integer class wraps a value of the primitive type int in an object.
    /// </summary>
    /// <author>Petr Kozler (A13B0359P)</author>
    public class Integer {

        /// <summary>
        /// A constant holding the maximum value an int can have, 2^31-1.
        /// </summary>
        public const int MAX_VALUE = 2147483647;

        /// <summary>
        /// A constant holding the minimum value an int can have, -2^31.
        /// </summary>
        public const int MIN_VALUE = -2147483648;

        private int v;

        /// <summary>
        /// Constructs a newly allocated Integer object that represents the specified
        /// int value.
        /// </summary>
        /// <param name="value">the value to be represented by the Integer object.
        /// </param>
        public Integer(int value) {
            v = value;
        }

        /// <summary>
        /// Returns the value of this Integer as an int.
        /// </summary>
        /// <returns>the numeric value represented by this object after conversion to
        /// type int.
        /// </returns>
        public int intValue() {
            return v;
        }

        /// <summary>
        /// Compares two Integer objects numerically.
        /// </summary>
        /// <param name="anotherInteger">the Integer to be compared.
        /// </param><returns>the value 0 if this Integer is equal to the argument Integer; a
        /// value less than 0 if this Integer is numerically less than the argument
        /// Integer; and a value greater than 0 if this Integer is numerically
        /// greater than the argument Integer (signed comparison).
        /// </returns>
        public int compareTo(Integer anotherInteger) {
            return compare(v, anotherInteger.v);
        }

        /// <summary>
        /// Compares two int values numerically.
        /// </summary>
        /// <param name="x">the first int to compare
        /// </param><param name="y">the second int to compare
        /// </param><returns>the value 0 if x == y; a value less than 0 if x < y; and a value greater than 0 if x
        /// > y
        /// </returns>
        public static int compare(int x, int y) {
            /* 1 if x is greater, 0 if x equals y, -1 otherwise
             (same as for a long type) */
            return (x < y) ? -1 : ((x == y) ? 0 : 1);
        }

        /// <summary>
        /// Compares this object to the specified object.
        /// </summary>
        /// <param name="obj">the object to compare with.
        /// </param><returns>true if the objects are the same; false otherwise.
        /// </returns>
        public bool equals(Object obj) {
            if (obj == null) {
                return false;
            }

            if (GetType() != obj.GetType()) {
                return false;
            }

            return v.Equals((obj as  Integer).v);
        }

        /// <summary>
        /// Returns a String object representing this Integer's value.
        /// </summary>
        /// <returns>a string representation of the value of this object in base 10.
        /// </returns>
        public String toString() {
            return Integer.toString(v);
        }

        /// <summary>
        /// Returns a String object representing the specified integer.
        /// </summary>
        /// <param name="i">an integer to be converted.
        /// </param><returns>a string representation of the argument in base 10.
        /// </returns>
        public static String toString(int i) {
            return i.ToString();
        }

        /// <summary>
        /// Parses the string argument as a signed decimal integer.
        /// </summary>
        /// <param name="s">a String containing the int representation to be parsed
        /// </param><returns>the integer value represented by the argument in decimal.
        /// </returns>
        public static int parseInt(String s) {
            return int.Parse(s.ToString());
        }
        
        public override string ToString() {
            return toString().ToString();
        }

    }

}

File: CPlusPlus/CppLibrary/Integer.cpp
#include "Integer.h"

#include <string>
#include <cstdlib>

const int32_t Integer::MAX_VALUE = 2147483647;
const int32_t Integer::MIN_VALUE = -2147483648;

/**
 * Constructs a newly allocated Integer object that represents the specified
 * int value.
 *
 * @param value the value to be represented by the Integer object.
 */
Integer::Integer(int32_t value) {
    v = value;
}

/**
 * Returns the value of this Integer as an int.
 *
 * @return the numeric value represented by this object after conversion to
 * type int.
 */
int32_t Integer::intValue() {
    return v;
}

/**
 * Compares two Integer objects numerically.
 *
 * @param anotherInteger the Integer to be compared.
 * @return the value 0 if this Integer is equal to the argument Integer; a
 * value less than 0 if this Integer is numerically less than the argument
 * Integer; and a value greater than 0 if this Integer is numerically
 * greater than the argument Integer (signed comparison).
 */
int32_t Integer::compareTo(Object *anotherInteger) {
    return Integer::compare(v, ((Integer *)anotherInteger)->v);
}

/**
 * Compares two int values numerically.
 *
 * @param x the first int to compare
 * @param y the second int to compare
 * @return the value 0 if x == y; a value less than 0 if x < y; and a value greater than 0 if x
 * > y
 */
int32_t Integer::compare(int32_t x, int32_t y) {
    // 1 if x is greater, 0 if x equals y, -1 otherwise
    return (x < y) ? -1 : ((x == y) ? 0 : 1);
}

/**
 * Compares this object to the specified object.
 *
 * @param obj the object to compare with.
 * @return true if the objects are the same; false otherwise.
 */
bool Integer::equals(Object *obj) {
    // testing another object reference for a NULL value
    if (obj == nullptr) {
        return false;
    }

    // testing object class equality
    if (sizeof(this) != sizeof(*obj)) {
        return false;
    }

    // testing object fields equality
    return (v == ((Integer *)obj)->v);
}

/**
 * Returns a String object representing this Integer's value.
 * 
 * @return a string representation of the value of this object in base 10.
 */
String *Integer::toString() {
    return toString(v);
}

/**
 * Returns a String object representing the specified integer.
 *
 * @param i an integer to be converted.
 * @return a string representation of the argument in base 10.
 */
String *Integer::toString(int32_t i) {
    return new String(std::to_string(i));
}

/**
 * Parses the string argument as a signed decimal integer.
 *
 * @param s a String containing the int representation to be parsed
 * @return the integer value represented by the argument in decimal.
 */
int32_t Integer::parseInt(String *s) {
    String *str = s->trim();
    int32_t value = strtol(str->_s().c_str(), nullptr, 10);
    delete str;
    
    return value;
}
